### ISSUE TITLE
[Request] Add baseUrl option

### ISSUE THREAD

**Issue by juanamd** [LonelyCpp/react-native-youtube-iframe#89]
Hi,

I was wondering if it would be possible to either:
-Enable an optional prop to set the 'baseUrl' property of the WebView's [source prop](https://github.com/react-native-webview/react-native-webview/blob/master/docs/Reference.md#source)
-Set (hardcode) the baseUrl to "https://youtube.com" as this solves the _"video unavailable"_ [issues](https://github.com/LonelyCpp/react-native-youtube-iframe/issues/5).

I only barely tested it on an Android emulator and it works for me but wasn't sure if this was intentionally omitted as an option for other reasons.

**Comment by LonelyCpp** [comment]
Hey, I've added a provision to specify `baseUrl` 
https://github.com/LonelyCpp/react-native-youtube-iframe/blob/093af584b38495ccd8215917bf7b5a03903ffd74/src/YoutubeIframe.js#L196

1. point your `package.json` dependency to the master branch 

    `"react-native-youtube-iframe": "https://github.com/lonelyCpp/react-native-youtube-iframe#master"`

2. and specify it like this - 
```
     webViewProps={{
        source: { baseUrl: "https://youtube.com" },
      }}
```

**Comment by LonelyCpp** [comment]
let me know if it works for you

**Comment by juanamd** [comment]
Tried using master but fails with: 
"'rm' is not recognized as an internal or external command,"

I'm on Windows and it seems that [this ](https://github.com/LonelyCpp/react-native-youtube-iframe/commit/b2ee3ef8fd213550af9fab2e2eb8e5d0d0774207) commit is not entirely compatible with my setup.

**Comment by LonelyCpp** [comment]
ah damn, makes sense. I will patch this soon

**Comment by LonelyCpp** [comment]
Hi @juanamd, the master branch now uses [rimraf](https://www.npmjs.com/package/rimraf). can you try now?

**Comment by juanamd** [comment]
Now it works, thanks!
The only issue I see now is that some typedefs appear to be outdated as I get this error:

`Type '{ baseUrl: string; }' is not assignable to type 'WebViewSourceUri | WebViewSourceHtml | undefined'.
  Property 'html' is missing in type '{ baseUrl: string; }' but required in type 'WebViewSourceHtml'.`

**Comment by LonelyCpp** [comment]
closing this in favour of baseUrlOverride https://github.com/LonelyCpp/react-native-youtube-iframe/blob/f0b816e139eb1938ffe4b906550ed4ceefc4221f/src/YoutubeIframe.js#L35

This will also fix the `video unavailable` issues

**Comment by LonelyCpp** [comment]
>baseUrl: baseUrlOverride

ya this changes does make sense.

But I'm curious why you would want this since the hosted solution will solve all "video unavailable" problems

**Comment by juanamd** [comment]
Just being nitpicky mostly...
If the default host (GitHub pages I believe?) is down/unreachable then the videos will stop working.  Or maybe there is an update to the script which breaks something in the future...
I know, I know, both quite unlikely, but if I can have a local solution that works then it reduces the points of failure I guess.

**Comment by LonelyCpp** [comment]
gotcha

**Comment by LonelyCpp** [comment]
fixed here - https://github.com/LonelyCpp/react-native-youtube-iframe/commit/d4a28b148990c5431b1c092f611ae9eed2a97634

available on `v2.0.1`

**Comment by juanamd** [comment]
Great, thanks!
